crates/primitives/src/block.rs

```diff
@@ -405,3 +391,12 @@ impl fmt::Display for BlockNumberOrTag {
         }
     }
 }
+
+/// A block hash and an optional bool that defines if it's canonical
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct BlockHashCanonical {
+    /// A block hash
+    block_hash: H256,
+    /// Whether the block must be a canonical block
+    require_canonical: Option<bool>,
```

Comment:
this should link to the EIP and cite relevant section

Reply:
[Done!](https://github.com/paradigmxyz/reth/pull/1237/files#diff-644939f544eae3cce57fadf1b1bcc80c2147c7c0f198ef6830768111af6f846aR411)